subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DetachedConfigurationsIntegrationTest.groovy

```diff
@@ -126,9 +118,12 @@ class DetachedConfigurationsIntegrationTest extends AbstractIntegrationSpec {
             detached.outgoing.artifact(tasks.makeArtifact)
 
             task checkDependencies {
+                def depModuleNames = detached.incoming.resolutionResult.root.dependencies*.selected*.moduleVersion*.name
+                def artifactNames = detached.incoming.artifacts.artifacts.collect { it.file.name }
+
                 doLast {
-                    assert detached.resolvedConfiguration.getFirstLevelModuleDependencies().moduleName.contains('test')
-                    assert detached.resolvedConfiguration.resolvedArtifacts.collect { it.file.name }.contains("artifact.zip")
+                    assert depModuleNames.contains('test')
+                    assert artifactNames.contains("artifact.zip")
```

Comment:
Won't work: `cannot serialize object of type 'org.gradle.api.internal.artifacts.configurations.DefaultConfiguration$ConfigurationResolvableDependencies$LenientResolutionResult`.  Would have to make it CC incompatible, like the other tests now using custom tasks to avoid resolution during the configuration phase.